_posts/2021-05-19-#18418.md

```diff
@@ -0,0 +1,110 @@
+that knows our personal information or IP address), we might accidentally
+reveal how much money we have!
+
+* The Bitcoin Core wallet implements a few best-practice privacy techniques.
+  One is avoiding the reuse of addresses when creating an invoice or change
+address.  Another is grouping UTXOs into
+[OutputGroup](https://github.com/bitcoin-core-review-club/bitcoin/blob/4ac1adda9914d845aaea5804af4801ffec53c701/src/wallet/coinselection.h#L72)s
+by scriptPubKey and spending them together.
+
+* However, `OutputGroup`s can grow quite large - after all, anyone can send as
+  many transactions as they want to an address that we've published. It might
+not make sense to fund a 0.015BTC transaction by sweeping a group of 150 inputs
+worth 10BTC (not to mention the extra fees for all the unnecessary inputs).
+The `OUTPUT_GROUP_MAX_ENTRIES` constant limits the number of UTXOs per
+`OutputGroup`. Within `GroupOutputs()`, if a scriptPubKey exceeds this many, we
+batch them into multiple `OutputGroup`s with up to `OUTPUT_GROUP_MAX_ENTRIES`
+UTXOs each.
+
+* [PR#18418](https://github.com/bitcoin/bitcoin/pull/18418) increases
+  `OUTPUT_GROUP_MAX_ENTRIES` from 10 to 100. The number 100 was suggested
+during a previous review club,
+[#17824](https://bitcoincore.reviews/17824.html#l-339) This behavior change
+constitutes just one line (and some adjustments to the tests), but it is ripe
+with opportunities to explore how coin selection works.
+
+* For beginners, host recommends simply adding log statements and (don't forget
+  to re-compile) then re-running the tests (Hint:
+`test/functional/combine_logs.py` to see logs, and you can assert that your
+logs are printed by adding `with
+node.assert_debug_log(expected_msg=[your_log_statement])` to the functional
+test).  Some good tests to play around with are wallet\_avoidreuse.py and
+wallet\_groups.py.  [This doc](https://github.com/fjahr/debugging_bitcoin) is a
+great resource (also authored by fjahr)!  You can also tinker some of the
+constants (maybe poke around for off-by-one errors) and see if things break!
+
+* You may find some previous review clubs helpful:
+
+	- Review Club [#17824](/17824) discussed the `avoid_reuse` flag.
+	- Review Clubs [#17331](/17331) and [#17526](/17526) discussed coin
+	  selection.
+
+
+## Questions
+
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
+What was your review approach?
+
+2. What do the `avoid_reuse` wallet flag and `-avoidpartialspends` wallet
+options do?  Which of the following combinations are possible / make sense?
+
+	(A) `avoid_reuse = true, avoidpartialspends = true`
+
+	(B) `avoid_reuse = true, avoidpartialspends = false`
+
+	(C) `avoid_reuse = false, avoidpartialspends = true`
+
+	(D) `avoid_reuse = false, avoidpartialspends = false`
+
+
+3. If your wallet has 101 UTXOs of 0.01 BTC each and tries to send a payment of
```

Comment:
Does this need to state that the 101 UTXOs are to the same address?

Reply:
woops! good point